### DISCUSSION TITLE
How to set IF line padding only for blocks?

### DISCUSSION THREAD

**Discussion by thdoan** [eslint/eslint#17286] [Q&A] [answered]
I'm struggling to add a rule that enforces one line padding before and after IF statements, but only if the IF statement is a block.

Examples:

```
// OK
let flag = true;
if (flag) console.log(flag);
let n = 0;

// ERROR
let flag = true;
if (flag) {
  console.log(flag);
}
let n = 0;

// OK
let flag = true;

if (flag) {
  console.log(flag);
}

let n = 0;
```

Is it possible? I played around with the `padding-line-between-statements` rule, but the `if` option treats all IF statements the same regardless of whether they are blocks or non-blocks.

**Comment by mdjermanovic** [comment, marked as answer]
There's no option specifically for block IF statements, but you could use `"block-like"` to enforce padding before and after all block-like statements (block IF, block WHILE...).

Here's an [ESLint Playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgcGFkZGluZy1saW5lLWJldHdlZW4tc3RhdGVtZW50czogW1xuICAgIFwiZXJyb3JcIixcbiAgICB7IGJsYW5rTGluZTogXCJhbHdheXNcIiwgcHJldjogXCIqXCIsIG5leHQ6IFwiYmxvY2stbGlrZVwiIH0sXG4gICAgeyBibGFua0xpbmU6IFwiYWx3YXlzXCIsIHByZXY6IFwiYmxvY2stbGlrZVwiLCBuZXh0OiBcIipcIiB9XG5dKi9cblxuLy8gT0tcbmxldCBmbGFnMSA9IHRydWU7XG5pZiAoZmxhZzEpIGNvbnNvbGUubG9nKGZsYWcxKTtcbmxldCBuMSA9IDA7XG5cbi8vIEVSUk9SXG5sZXQgZmxhZzIgPSB0cnVlO1xuaWYgKGZsYWcyKSB7XG4gIGNvbnNvbGUubG9nKGZsYWcyKTtcbn1cbmxldCBuMiA9IDA7XG5cbi8vIE9LXG5sZXQgZmxhZzMgPSB0cnVlO1xuXG5pZiAoZmxhZzMpIHtcbiAgY29uc29sZS5sb2coZmxhZzMpO1xufVxuXG5sZXQgbjMgPSAwO1xuXG5cbiIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnsiZXM2Ijp0cnVlfX19) with your example code.
